docs/node-mixin/lib/prom-mixin.libsonnet

```diff
@@ -0,0 +1,353 @@
+local grafana = import 'github.com/grafana/grafonnet-lib/grafonnet/grafana.libsonnet';
+local dashboard = grafana.dashboard;
+local row = grafana.row;
+local prometheus = grafana.prometheus;
+local template = grafana.template;
+local graphPanel = grafana.graphPanel;
+local grafana70 = import 'github.com/grafana/grafonnet-lib/grafonnet-7.0/grafana.libsonnet';
+local gaugePanel = grafana70.panel.gauge;
+
+{
+
+  new(config=null, platform=null):: {
+
+    local prometheusDatasourceTemplate = {
+      current: {
+        text: 'default',
+        value: 'default',
+      },
+      hide: 0,
+      label: 'Data Source',
+      name: 'datasource',
+      options: [],
+      query: 'prometheus',
+      refresh: 1,
+      regex: '',
+      type: 'datasource',
+    },
+
+    local instanceTemplatePrototype =
+      template.new(
+        'instance',
+        '$datasource',
+        '',
+        refresh='time',
+        label='Instance',
+      ),
+    local instanceTemplate =
+      if platform == 'Darwin' then
+        instanceTemplatePrototype
+        { query: 'label_values(node_uname_info{%(nodeExporterSelector)s, sysname="Darwin"}, instance)' % config }
```

Comment:
I think instead of having a hardcoded sysname label we should have a separate nodeExporterSelectorDarwin and nodeExporterSelectorLinux

Reply:
in config.libsonnet?

Reply:
Just not sure when it might be required to change those?

Reply:
People can choose whatever label they want to distinguish between darwin and linux nodes, so it should be made configuratable. Yes in config.libsonnet, smae as nodeExporterSelector.

Reply:
In this case - `sysname` is fixed label specifically from node_uname_info metric. This filter is only applied to resolve $instance grafana variable and not to other queries.  I'm just not sure what else could be used here?